lib/ansible/module_utils/scaleway.py

```diff
@@ -21,6 +22,30 @@ def __init__(self, message):
         self.message = message
 
 
+# Specify a complete Link header, for validation purposes
+R_LINK_HEADER = r'''<[^>]+>;\srel="(first|previous|next|last)"
+    (,<[^>]+>;\srel="(first|previous|next|last))*"'''
+# Specify a single relation, for iteration and string extraction purposes
+R_RELATION = r'<(?P<target_IRI>[^>]+)>; rel="(?P<relation>first|previous|next|last)"'
+
+
+def parse_pagination_link(header):
+    if re.match(R_LINK_HEADER, header, re.VERBOSE):
```

Comment:
Could you have a syntax close to : 

```
if not re.match():
    raise ...
```

That we don't have as much indentation levels.